### PR TITLE
Feat/all deals

### PR DESCRIPTION
**Pull request by brandonseydel**
## What was done

All deals page was completed.
Removed deprecated node-sass compiler with current.
Refactor events so they are in one place for type checking.
Currency formatter was added with a optional decimal places param.

#### After

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -107,9 +107,9 @@
     "json-loader": "^0.5.7",
     "mini-css-extract-plugin": "^2.5.3",
     "minimatch": "^3.0.4",
-    "node-sass": "^4.14.1",
+    "sass": "^1.49.9",
+    "sass-loader": "^12.6.0",
     "promise-polyfill": "^8.1.3",
-    "sass-loader": "^12.4.0",
```
Comment (dkent600) [inline package.json:112]: why the sass changes?

File: src/dealDashboard/discussionThread/singleComment/singleComment.ts [hunk 2]
```diff
@@ -26,42 +16,53 @@ export class SingleComment {
     down: false,
   };
 
+  @bindable public author: string;
```
Comment (dkent600) [inline src/dealDashboard/discussionThread/singleComment/singleComment.ts:19]: I assume this is not really your change.  Some kind of merging problem?

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/dealDashboard/discussionThread/singleComment/singleComment.ts (hunk 2 above)
Comment (brandonseydel) [reply src/dealDashboard/discussionThread/singleComment/singleComment.ts:19]: There is nothing changed in here just ordered and bindables made public to represent what they actually are.

File: package.json (hunk 1 above)
Comment (brandonseydel) [reply package.json:112]: The node-sass library has been deprecated for over a year and actually requires additional software to run.  The sass package is its predecessor.  There should be no differences other than "sass" package has new and up to date features from the lang itself, requires no third party tooling/downloads (at least for windows), and is maintained through node versions.

You can see their warning and suggestion below.
https://github.com/sass/node-sass

Comment (dkent600) [reply package.json:112]: cool, and thanks, but our practice is that changes unrelated to what is being addressed by a PR should be done in a separate PR.  Can you please create a ticket for this and put it in the Triage column (the default "pipeline") of the kanban, and back the change out of this PR?

File: src/dealDashboard/discussionThread/singleComment/singleComment.ts (hunk 2 above)
Comment (dkent600) [reply src/dealDashboard/discussionThread/singleComment/singleComment.ts:19]: cool, and thanks, but our practice is that changes unrelated to what is being addressed by a PR should be done in a separate PR.  Can you please create a ticket for this and put it in the Triage column (the default "pipeline") of the kanban, and back the change out of this PR?

File: src/app.ts [hunk 3]
```diff
@@ -39,7 +40,7 @@ export class App {
   intervalId: any;
 
   errorHandler = (ex: unknown): boolean => {
-    this.eventAggregator.publish("handleException", new EventConfigException("Sorry, an unexpected error occurred", ex));
+    this.eventAggregator.publish(EventType.HandleException, new EventConfigException("Sorry, an unexpected error occurred", ex));
```
Comment (dkent600) [inline src/app.ts:43]: our practice is that changes unrelated to what is being addressed by a PR should be done in a separate PR.  Can you please create a ticket for the `EventType` enum and put it in the Triage column (the default "pipeline") of the kanban, and back the change out of this PR?
